File: server/internal/model/challenge.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Challenge struct {
	ID           uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	UserID       uuid.UUID `gorm:"type:uuid;index;not null"`
	User         User      `gorm:"foreignKey:UserID"`
	Type         string    `gorm:"not null"`
	TargetValue  float64   `gorm:"not null"`
	CurrentValue float64   `gorm:"default:0"`
	StartDate    time.Time `gorm:"not null"`
	EndDate      time.Time `gorm:"not null"`
	IsCompleted  bool      `gorm:"default:false"`
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

File: mobile/src/models/Challenge.ts
import { Model } from "@nozbe/watermelondb";
import { field } from "@nozbe/watermelondb/decorators";

import type { ChallengeType } from "../constants/enums";

export class Challenge extends Model {
  static table = "challenges";

  @field("type") type!: ChallengeType;
  @field("target_value") targetValue!: number;
  @field("current_value") currentValue!: number;
  @field("start_date") startDate!: number;
  @field("end_date") endDate!: number;
  @field("is_completed") isCompleted!: boolean;
  @field("created_at") createdAt!: number;
  @field("updated_at") updatedAt!: number;
}
